packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/read.rs

```diff
@@ -5,86 +5,67 @@ use postgres_types::ToSql;
 use serde::Deserialize;
 use tokio_postgres::GenericClient;
 
-use crate::{
-    ontology::types::uri::VersionedUri,
-    store::{
-        crud, crud::AllLatest, postgres::ontology::OntologyDatabaseType, AsClient, PostgresStore,
-        QueryError,
-    },
+use crate::store::{
+    crud::Read,
+    postgres::ontology::OntologyDatabaseType,
+    query::{OntologyQuery, OntologyVersion},
+    AsClient, PostgresStore, QueryError,
 };
 
-#[async_trait]
-impl<'i, C: AsClient, T> crud::Read<'i, &'i VersionedUri, T> for PostgresStore<C>
-where
-    for<'de> T: OntologyDatabaseType + Deserialize<'de>,
-{
-    type Output = T;
-
-    async fn get(&self, uri: &'i VersionedUri) -> Result<Self::Output, QueryError> {
-        let version = i64::from(uri.version());
-        // Generally bad practice to construct a query without preparation, but it's not possible to
-        // pass a table name as a parameter and `T::table()` is well-defined, so this is a safe
-        // usage.
-        let row = self
-            .as_client()
-            .query_one(
-                &format!(
-                    r#"
-                    SELECT schema
-                    FROM {}
-                    WHERE version_id = (
-                        SELECT version_id
-                        FROM ids
-                        WHERE base_uri = $1 AND version = $2
-                    )
-                    "#,
-                    T::table()
-                ),
-                &[uri.base_uri(), &version],
-            )
-            .await
-            .into_report()
-            .change_context(QueryError)
-            .attach_printable_lazy(|| uri.clone())?;
-
-        serde_json::from_value(row.get(0))
-            .into_report()
-            .change_context(QueryError)
-    }
+fn parameter_list<const N: usize>(list: [&(dyn ToSql + Sync); N]) -> [&(dyn ToSql + Sync); N] {
```

Comment:
I suppose this is to get rid of the `[] as [&(dyn ToSql + Sync); 0]` we had previously in zero-arg queries?

Reply:
Yes, and it's also useful to convert other types to convert to the desired type. It's convenient to call it when interacting with `query_raw`.